LibreNMS/OS/Lcos.php

```diff
@@ -0,0 +1,308 @@
+<?php
+/**
+ * Lcos.php
+ *
+ * Lancom LCOS
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * @package    LibreNMS
+ * @link       http://librenms.org
+ * @copyright  2019 Vitali Kari
+ * @author     Vitali Kari <vitali.kari@gmail.com>
+ */
+
+namespace LibreNMS\OS;
+
+use LibreNMS\Device\WirelessSensor;
+use LibreNMS\Interfaces\Discovery\Sensors\WirelessFrequencyDiscovery;
+use LibreNMS\Interfaces\Polling\Sensors\WirelessFrequencyPolling;
+use LibreNMS\Interfaces\Discovery\Sensors\WirelessCapacityDiscovery;
+use LibreNMS\Interfaces\Discovery\Sensors\WirelessNoiseFloorDiscovery;
+use LibreNMS\Interfaces\Discovery\Sensors\WirelessPowerDiscovery;
+use LibreNMS\Interfaces\Discovery\Sensors\WirelessCcqDiscovery;
+use LibreNMS\Interfaces\Discovery\Sensors\WirelessRateDiscovery;
+use LibreNMS\Interfaces\Discovery\Sensors\WirelessRssiDiscovery;
+use LibreNMS\OS;
+
+class Lcos extends OS implements
+    WirelessFrequencyDiscovery,
+    WirelessFrequencyPolling,
+    WirelessCapacityDiscovery,
+    WirelessNoiseFloorDiscovery,
+    WirelessPowerDiscovery,
+    WirelessCcqDiscovery,
+    WirelessRateDiscovery,
+    WirelessRssiDiscovery
+{
+    /**
+     * Convert String to decimal encoded string notation
+     *
+     * @param string
+     * @return decimal encoded OID string
+     */
+    private function strToDecOid($index)
+    {
+        for ($i = 0, $j = strlen($index); $i < $j; $i++) {
+                $dec_index[] = ord($index{$i});
+        }
+        return implode('.', $dec_index);
+    }
+
+    /**
+     * Convert MAC String to decimal encoded string notation
+     *
+     * @param colon separated string (MAC address notation)
+     * @return decimal encoded OID string
+     */
+    private function macToDecOid($mac)
```

Comment:
I think this one already exists somewhere in LibreNMS ... don't remember where at all, but I saw it (and used it) before.

Reply:
found some places that uses similar code, but nowhere a function for that.
Maybe this is the right time to do this now, I'm not sure if a protected function in LibreNMS/OS.php or a just drop it to includes/functions.php would be the right way.
Or we leave that actually as it. :-)

Reply:
Found it : 
```
/opt/librenms/LibreNMS/Util/Rewrite.php:    public static function oidMac($mac)
```
So you could use this one and remove your new Implementation. Rewrite.php is where this kind of function is added in the newer architecture. includes/functions.php will eventually be deprecated.